### ISSUE TITLE
False Positive for Unused -> "Parameter/variable is not used inspection" with objects by reference

### ISSUE THREAD

**Issue by tfettig01** [kalessil/phpinspectionsea#765]
"Analyzes functions and methods control flow, reports unused variables and potentially unused variables (such variables are used in write context only)."

A simplified example.
```php
function add(\ArrayObject $Foo, string $string)
{
    // False positive below
    $Foo[] = $string;
}

$Foo   = new ArrayObject();
$Foo[] = 'one';
add($Foo, 'two');
var_dump($Foo);
exit;
```

**Comment by kalessil** [comment]
@tfettig01, I'm in doubt here. Shouldn't `$Foo` be returned then as result?

@rentalhost @funivan : your thoughts?

**Comment by funivan** [comment]
Hello. I think this is definitely false positive. 
Maybe this can be fixed in the following manner:
0. Variable passed as function argument
1. variable is an object 
2. write operation does not require any read operations

**Comment by rentalhost** [comment]
`$foo` don't need be returned here because it is an object. The method will affect the contents because of `$foo[]` (once that it is an `ArrayObject`). So it is a false-positive.

**Comment by tfettig01** [comment]
> @tfettig01, I'm in doubt here. Shouldn't $Foo be returned then as result?

False positive? Yes. 
Bad code? A definite maybe.  Arguments could be made either way.

I'm seeing you took the issue as a false-positive.  Thank you.

**Comment by kalessil** [comment]
Fixed
